compiler/lookups.nim

```diff
@@ -170,19 +170,30 @@ iterator allSyms*(c: PContext): (PSym, int, bool) =
   # really iterate over all symbols in all the scopes. This is expensive
   # and only used by suggest.nim.
   var isLocal = true
+  # Track seen symbols so we don't duplicate them.
+  # The int is for the symbols name, and line info is
+  # to be able to tell apart symbols with same name but on different lines
+  var seen = initHashSet[(TLineInfo, int)]()
```

Comment:
Why does TLineInfo matter here? We should only produce the first unique *name* as it's for typo corrections.

Reply:
An example is if the user imports `options` and `httpclient`.
If they mistype `get` then it would be good to show them that there exists both a `get` for options and for httpclient.

Reply:
We also need to return all symbols so `suggest.nim` can go through the symbols and match the correct one.

Reply:
But then this doesn't fix the issue that it suggests 10x `shl` etc

Reply:
I could group them up into their modules to stop repeated suggestions?
e.g.
```nim
proc `shl`() = discard
proc shout() = discard

sh
```
Would give output
```
/tmp/test.nim(4, 1) Error: undeclared identifier: 'sh'
candidates; see '--spellSuggest':
 'shl':
    - /home/jake/.choosenim/toolchains/nim-#devel/lib/system/arithmetics.nim (first seen 158:6)
    - /tmp/test.nim (first seen (1, 6)
 'shout': 
    - declared in test.nim (first seen 2:6)
```

This way `nimsuggest` still gets all the results and the console isn't spammed for every declaration of a symbol

Reply:
Just use different code paths for the typo checker and nimsuggest. I agree that nimsuggest must get all the results.

Reply:
Now separate code paths. `nimsuggest` uses an iterator that builds on existing `allSyms` but only yields unique symbols (by line and name). The typo checker now skips symbols that are the same as the previous (They are sorted so its fairly efficient).

Remove line info from the typo suggestions. Since only the first declaration is shown it would be a bit misleading to the user if they couldn't see the other declarations ("Why is it only showing get in `options.nim` when I know its declared here also?" 

The output is a lot cleaner now. In an empty file if you type "sh" the output is like so
```
test.nim(1, 1) Error: undeclared identifier: 'sh'
candidates (edit distance, scope distance); see '--spellSuggest': 
 (1, 2): 'shl'
 (1, 2): 'shr'
```